src/gui/userstatus.cpp

```diff
@@ -30,16 +30,37 @@ Q_LOGGING_CATEGORY(lcUserStatus, "nextcloud.gui.userstatus", QtInfoMsg)
 
 UserStatus::UserStatus(QObject *parent)
     : QObject(parent)
-    , _message("")
 {
 }
 
 UserStatus::Status UserStatus::stringToEnum(const QString &status) const 
 {
+    // it needs to match the Status enum
+    const QHash<QString, Status> preDefinedStatus{{"online", Online},
+                                               {"dnd", DoNotDisturb}, //DoNotDisturb
+                                               {"away", Away},
+                                               {"offline", Offline},
+                                               {"invisible", Invisible}};
+    
     // api should return invisible, dnd,... toLower() it is to make sure 
     // it matches _preDefinedStatus, otherwise the default is online (0)
-    const auto statusEnum = _preDefinedStatus.value(status.isEmpty()? "online" : status.toLower(), 0);
-    return static_cast<Status>(statusEnum);
+    const auto statusKey = status.isEmpty() ? QStringLiteral("online") : status.toLower();
+    return preDefinedStatus.value(statusKey, Online);
+}
+
+QString UserStatus::enumToUserString(Status status) const 
+{
+    switch (status) {
```

Comment:
@camilasan I am still wondering why we can't just use `QString statusStrings[Status::Count];` and then just `return statusStrings[status];` ? :)

Reply:
Because you'd be over-thinking against the enum as it is today. Add a new enum value in the middle and then you got a subtle bug which is not necessarily easy to track down as the string can travel quite a bit until it's consumption point.

Using enum values as indices for arrays is a dangerous business in maintenance. There's a reason for enum class. ;-)

Reply:
@er-vin Alrighty then. Never thought about that, but,  it should also be simple to check that the enum index is within bounds and has a corresponding `QString` assigned to it. Now, if we are talking about `enum class`, why not make that `enum Status` an `enum class Status` instead, @camilasan ? :)

Reply:
The response from the server always send e.g. "dnd" and "online" and I am capitalizing the string. Also following Kevin's suggestion in the previous PR. Or I am missing something here...

Reply:
Somehow I missed Kevin's reply.
ok, enum Class it is then.